cluster-creation-cli/README.md

```diff
@@ -0,0 +1,163 @@
+
+## Basic syntax
+
+The command should be named `create cluster`, hence the entire command, without any flags, would be:
+
+    kubectl gs create cluster
+
+The most important flags would be:
+
+- `--provider`: (required) provider name. For example: capa, capvcd, capz, eks.
+  - TODO: How do we restrict and validate the provider list per installation?
+- `--config`: Path to a config file (optional). Can be used multiple times. See "Configuration layering" below. As an alternative, or in addition, configuration can be passed via standard input.
+- `--app-version`: the cluster-PROVIDER app version to use. If not specified, the latest version is used.
+- `--dry-run`: only validate the configuration and print the resulting manifest.
+- `--output`: file path to write the result to. If not specified, resources are written to the management cluster. The option `STDOUT` can be used to print the result to the console.
+- `--format`: This flag can be used to specify the format of the output.
+  - Default is `manifest` for a Kubernetes manifest that includes an App resources and one or several ConfigMap resources.
+  - `config` for merged configuration YAML.
+  - `command` for a self-contained representation of the command just executed, as a one-liner.
+- `--set`: Override a configuration value. This flag can be used multiple times. The format is `path.to.property=value`.
+  - Note: `--set` can only be used with scalar values (string, number, boolean, integer, etc.). It cannot be used to override an array or object.
+
+TODO:
+
+- How should the resource **namespace** be specified? It could be derived from the organization name included in the cluster config, however that would require logical understanding of the config, which may differ over time and between providers.
+
+### Examples
+
+Ex. 1: Simple cluster creation based on a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml
+
+Ex. 2: Simple cluster creation based on standard input
+
+    kubectl gs create cluster --provider capa <<EOF
+    global:
+      metadata:
+        servicePriority: lowest
+        name: test01
+        organization: testorg
+        description: Just a test cluster
+    EOF
+
+While ex. 1 appears simpler and allows for easier re-use of a file-based configuration for a single user, ex. 2 has the benefit of being completely self-contained. It's a pure one-liner without the need for a separate file. This way it's easy to share the command, or copy the command from a website or web UI and paste it into a terminal.
+
+Ex. 3: Overwrite a configuration value
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --set global.metadata.name=test02
+
+Ex. 4: Dry run
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --dry-run
+
+Ex. 5: Write the result to a file
+
+    kubectl gs create cluster --provider capa --config myconfig.yaml --output manifest.yaml
```

Comment:
As I commented above, I think this would be better to make use of the `--dry-run` and redirect to a file like with other kubectl commands. e.g.
```
kubectl gs create cluster --provider capa --config myconfig.yaml --dry-run > manifest.yaml
```

Reply:
This is how it's specified now.